java/client/src/main/java/compatibility/clients/jedis/Jedis.java

```
@@ -387,4 +408,1394 @@ public Set<String> keys(String pattern) {
+            return Arrays.asList(result);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("SORT operation failed", e);
+        }
+    }
+
+    /**
+     * Sort the elements in a list, set, or sorted set with options.
+     *
+     * @param key the key
+     * @param sortingParameters sorting parameters (BY, LIMIT, GET, ASC/DESC, ALPHA)
+     * @return the sorted elements
+     */
+    public List<String> sort(String key, String... sortingParameters) {
+        checkNotClosed();
+        try {
+            if (sortingParameters.length == 0) {
+                // Simple sort without options
+                String[] result = glideClient.sort(key).get();
+                return Arrays.asList(result);
+            } else {
+                // Parse Jedis-style parameters into SortOptions
+                SortOptions sortOptions = parseSortParameters(sortingParameters);
+                String[] result = glideClient.sort(key, sortOptions).get();
+                return Arrays.asList(result);
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("SORT operation failed", e);
+        }
+    }
+
+    /**
+     * Parse Jedis-style sort parameters into GLIDE SortOptions.
+     *
+     * @param params the Jedis-style parameters (BY, LIMIT, GET, ASC/DESC, ALPHA)
+     * @return SortOptions object
+     */
+    private SortOptions parseSortParameters(String[] params) {
+        SortOptions.SortOptionsBuilder builder = SortOptions.builder();
+
+        for (int i = 0; i < params.length; i++) {
+            String param = params[i].toUpperCase();
+
+            switch (param) {
+                case "BY":
+                    if (i + 1 < params.length) {
+                        builder.byPattern(params[++i]);
+                    }
+                    break;
+
+                case "LIMIT":
+                    if (i + 2 < params.length) {
```

If there aren't enough parameters, does Jedis report an error? This current implementation just silently drops the modifier.

---

Optional modifiers are only added if there is some value for them. Most modifiers in redis commands are optional and are only added if we need to pass some valid value with them.